### PR TITLE
Avoid out of bound InsertRangeFrom exception in MergingSortedBlockInputStream

### LINKED ISSUE #8438: InsertRangeFrom out of bound in MergingSortedBlockInputStream
## Bug Report
commit hash: cf0e9affe7191b428a25142a91f240bce42823a1

[MPPTask.cpp:636] [\"task running meets error: Code: 12, e.displayText() = DB::Exception: Parameters are out of bound in ColumnString::insertRangeFrom method, start=0, length=18446744073709551563, src.size()=1, e.what() = DB::Exception, Stack trace:
       0x1ee1ade\\tDB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) [tiflash+32381662]\
                \\tdbms/src/Common/Exception.h:46\
       0x7fc3c7c\\tDB::ColumnString::insertRangeFrom(DB::IColumn const&, unsigned long, unsigned long) [tiflash+133971068]\
                \\tdbms/src/Columns/ColumnString.cpp:84\
       0x20a0a9b\\tDB::IColumn::cut(unsigned long, unsigned long) const [tiflash+34212507]\
                \\tdbms/src/Columns/IColumn.h:123\
       0x7c151e3\\tDB::MergingSortedBlockInputStream::readImpl() [tiflash+130109923]\
                \\tdbms/src/DataStreams/MergingSortedBlockInputStream.cpp:131\
       0x77cc995\\tDB::IProfilingBlockInputStream::read(DB::PODArray<unsigned char, 4096ul, Allocator<false>, 15ul, 16ul>*&, bool) [tiflash+125618581]\
                \\tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:82\
       0x77cc695\\tDB::IProfilingBlockInputStream::read() [tiflash+125617813]\
                \\tdbms/src/DataStreams/IProfilingBlockInputStream.cpp:48\
       0x8879033\\tDB::MergeSortTransformOp::executeIOImpl() [tiflash+143102003]\
                \\tdbms/src/Operators/MergeSortTransformOp.cpp:210\
       0x89356b4\\tDB::Operator::executeIO() [tiflash+143873716]\
                \\tdbms/src/Operators/Operator.cpp:81\
       0x886717a\\tDB::PipelineTaskBase::runExecuteIO() [tiflash+143028602]\
                \\tdbms/src/Flash/Pipeline/Schedule/Tasks/PipelineTaskBase.h:88\
       0x8956e7a\\tDB::Task::executeIO() [tiflash+144010874]\
                \\tdbms/src/Flash/Pipeline/Schedule/Tasks/Task.cpp:140\
       0x1ea1115\\tDB::TaskThreadPool<DB::IOImpl>::loop(unsigned long) [tiflash+32117013]\
                \\tdbms/src/Flash/Pipeline/Schedule/ThreadPool/TaskThreadPool.cpp:61\
       0x1ea1956\\tvoid* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (DB::TaskThreadPool<DB::IOImpl>::*)(unsigned long), DB::TaskThreadPool<DB::IOImpl>*, unsigned long> >(void*) [tiflash+32119126]\
                \\t/usr/local/bin/../include/c++/v1/thread:291\
  0x7f8d3d761802\\tstart_thread [libc.so.6+653314]\
  0x7f8d3d701314\\t__clone [libc.so.6+258836]\"] [source=\"MPP<gather_id:1, query_ts:1701237175718694001, local_query_id:10, server_id:782, start_ts:445969113639813121,task_id:19>\"] [thread_id=265]","namespace":"endless-htap-spill-to-disk-tps-4860765-1-214"}

### PR DESCRIPTION
**Pull request by yibin87**
### What problem does this PR solve?

Issue Number: close #8438 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/DataStreams/MergingSortedBlockInputStream.cpp [hunk 1]
```diff
@@ -219,11 +219,18 @@ void MergingSortedBlockInputStream::merge(MutableColumns & merged_columns, std::
 
                 if (limit && total_merged_rows + merged_rows > limit)
```
[the remark this answers is not in the document]
Comment (SeaRise) [reply dbms/src/DataStreams/MergingSortedBlockInputStream.cpp:220]: +1, It seems that `total_merged_rows` cannot be greater than `limit`.

[the remark this answers is not in the document]
Comment (yibin87) [reply dbms/src/DataStreams/MergingSortedBlockInputStream.cpp]: Good catch, I'll update the code.

**Review by xzhangxian1008 — APPROVED** [review]
LGTM

**Review by SeaRise — APPROVED** [review]
lgtm

**Pull request merged**